build-tools/src/integTest/groovy/org/elasticsearch/gradle/plugin/StableBuildPluginPluginFuncTest.groovy

```diff
@@ -0,0 +1,127 @@
+ */
+
+package org.elasticsearch.gradle.plugin
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+
+import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest
+import org.gradle.testkit.runner.TaskOutcome
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Collectors
+
+class StableBuildPluginPluginFuncTest extends AbstractGradleFuncTest {
+
+    def setup() {
+        // underlaying TestClusterPlugin and StandaloneRestIntegTestTask are not cc compatible
+        configurationCacheCompatible = false
+    }
+
+    def "can build stable plugin properties"() {
+        given:
+        buildFile << """plugins {
+                id 'elasticsearch.stable-esplugin'
+            }
+
+            version = '1.2.3'
+
+            esplugin {
+                name = 'myplugin'
+                description = 'test plugin'
+            }
+            """
+
+        when:
+        def result = gradleRunner(":pluginProperties").build()
+        def props = getPluginProperties()
+
+        then:
+        result.task(":pluginProperties").outcome == TaskOutcome.SUCCESS
+        props.get("classname") == null
+
+        props.get("name") == "myplugin"
+        props.get("version") == "1.2.3"
+        props.get("description") == "test plugin"
+        props.get("modulename") == ""
+        props.get("java.version") == Integer.toString(Runtime.version().feature())
+        props.get("elasticsearch.version") == VersionProperties.elasticsearchVersion.toString()
+        props.get("extended.plugins") == ""
+        props.get("has.native.controller") == "false"
+        props.size() == 8
+
+    }
+
+    def "can scan and create named components file"() {
+        given:
+        println testProjectDir.root
+        File jarFolder = new File(testProjectDir.root, "jars")
+        jarFolder.mkdirs()
+
+        buildFile << """plugins {
+                id 'elasticsearch.stable-esplugin'
+            }
+
+            version = '1.2.3'
+
+            esplugin {
+                name = 'myplugin'
+                description = 'test plugin'
+            }
+
+            dependencies {
+                implementation files('${pluginApiJar(jarFolder).absolutePath}')
+                implementation files('${extensibleApiJar(jarFolder).absolutePath}')
+            }
+
+            """
+
+        file("src/main/java/org/acme/A.java") << """
+            package org.acme;
+
+            import org.elasticsearch.plugin.api.NamedComponent;
+            import org.elasticsearch.plugin.scanner.test_classes.ExtensibleClass;
+
+            @NamedComponent(name = "componentA")
+            public class A extends ExtensibleClass {
+            }
+        """
+
+
+        when:
+        def result = gradleRunner(":assemble").build()
+        Path namedComponents = file("build/generated-named-components/named_components.json").toPath();
```

Comment:
Just a side note: Using JsonSlurper API from groovy is probably easier to read here. 

See https://docs.groovy-lang.org/next/html/gapi/groovy/json/JsonSlurper.html